app/javascript/controllers/nextflow/samplesheet_controller.js

```diff
@@ -762,6 +764,29 @@ export default class extends Controller {
     this.element.lastElementChild.remove();
   }
 
+  #updateMetadataColumnHeaderNames() {
+    // Update the values for the fields under 'The column header names of the metadata columns'
+
+    let metadataSamplesheetColumns = document.querySelectorAll(
+      '[id^="field-metadata_"]',
+    );
+
+    metadataSamplesheetColumns.forEach(function (metadataSamplesheetColumn) {
+      const columnName = metadataSamplesheetColumn.getAttribute(
+        "data-metadata-header",
+      );
+      const metadataField = metadataSamplesheetColumn.value;
+
+      let metadataParameter = document.querySelector(
```

Comment:
Suggestion (blocking): Instead of using `document` should be limiting the selector to the nextflow component element.